force-app/main/default/classes/ListDonation.cls

```diff
@@ -0,0 +1,111 @@
+        //public String paymentMethod { get; set; }
+    }
+
+    public static void getDonationHistory(){
+        List<OpportunityContactRole> donations = [SELECT Id, Opportunity.Name, Opportunity.CloseDate, Opportunity.Amount, Contact.Email, Contact.Name, Role, isPrimary FROM OpportunityContactRole WHERE Opportunity.StageName = 'Closed Won'];
+        List<DonationHistory> result = verifyCreditCondition(donations);
+    }
+
+    public static List<DonationHistory> verifyCreditCondition(List<OpportunityContactRole> donations){
+        List<DonationHistory> donationHistory = new List<DonationHistory>();
+        List<DonationHistory> hardCredit = verifyHardCreditCondition(donations);
+        donationHistory.addAll(hardCredit);
+
+        List<DonationHistory> softCredit = verifySoftCreditCondition(donations);
+        donationHistory.addAll(softCredit);
+
+        List<Id> donationsId = filterVerifiedIds(donationHistory, donations);
+        List<DonationHistory> partialSoftCredit = verifyPartialSoftCreditConsition(donationsId, donations);
+        donationHistory.addAll(partialSoftCredit);
+        System.debug(donationHistory);
+        return donationHistory;
+    }
+    
+    public static List<DonationHistory> verifyHardCreditCondition(List<OpportunityContactRole> donations){
+        List<DonationHistory> hardCreditList = new List<DonationHistory>();
+        for(OpportunityContactRole opp: donations){
+            if (opp.isPrimary) {
+                DonationHistory hardCredit = new DonationHistory();
+                hardCredit.id = opp.Id;
+                hardCredit.name = opp.Opportunity.Name;
+                hardCredit.closeDate = opp.Opportunity.CloseDate;
+                hardCredit.role = opp.Role;
+                hardCredit.amount = opp.Opportunity.Amount;
+                hardCredit.creditType = HARD_CREDIT;
+                hardCreditList.add(hardCredit);
+            }
+        }
+        return hardCreditList;
+    }
+    
+    public static List<DonationHistory> verifySoftCreditCondition(List<OpportunityContactRole> donations){
+        //with the list of donations verify if the Role match the filter rule value on the NPSP_ClosedWon_Opps_SC filter group
+        Filter_Rule__mdt roles = [SELECT Constant__c FROM Filter_Rule__mdt WHERE QualifiedApiName = 'NPSP_OppsSoftCreditRoles'];
+        List<String> softCreditRoles = roles.Constant__c.split(';');
+        List<DonationHistory> softCreditList = new List<DonationHistory>();
+        for(OpportunityContactRole opp: donations){
+            if (softCreditRoles.contains(opp.Role)) {
+                DonationHistory softCredit = new DonationHistory();
+                softCredit.id = opp.Id;
+                softCredit.name = opp.Opportunity.Name;
+                softCredit.closeDate = opp.Opportunity.CloseDate;
+                softCredit.role = opp.Role;
+                softCredit.amount = opp.Opportunity.Amount;
+                softCredit.creditType = HARD_CREDIT;
```

Comment:
I am confused as if to this should be `softCredit.creditType = SOFT_CREDIT;`

Reply:
fixed